cmd/daemon/controls.go

```diff
@@ -22,7 +22,14 @@ func (p *AppPlayer) prefetchNext() {
 		return
 	}
 
-	nextId := librespot.SpotifyIdFromUri(next.Uri)
+	// TODO: the same logic is present in ContextTrackToProvidedTrack and should
+	// probably be generalized.
+	var nextId librespot.SpotifyId
+	if next.Uri != "" {
+		nextId = librespot.SpotifyIdFromUri(next.Uri)
+	} else {
+		nextId = librespot.SpotifyIdFromGid(librespot.SpotifyIdTypeTrack, next.Gid)
```

Comment:
Are we sure `SpotifyIdTypeTrack` is always the correct type? Episode might end up here too. A call to `InferSpotifyIdTypeFromContextUri` might be needed.

Reply:
I'm not. I'll investigate this a bit more and hope to find some reliable reproducer so I can test this with some podcasts. Until then, I'm not fully convinced yet this is a correct fix.
Thanks for the pointer towards `InferSpotifyIdTypeFromContextUri`!